addons/website_portal/controllers/main.py

```diff
@@ -105,14 +105,11 @@ def details_form_validate(self, data):
 
         # vat validation
         if data.get("vat") and hasattr(request.env["res.partner"], "check_vat"):
-            if request.website.company_id.sudo().vat_check_vies:
-                # force full VIES online check
-                check_func = request.env["res.partner"].vies_vat_check
-            else:
-                # quick and partial off-line checksum validation
-                check_func = request.env["res.partner"].simple_vat_check
+            if data.get("country_id"):
+                data["vat"] = request.env["res.partner"].fix_eu_vat_number(int(data.get("country_id")), data.get("vat"))
+            check_func = request.website.company_id.vat_check_vies and request.env["res.partner"].vies_vat_check or request.env["res.partner"].simple_vat_check
```

Comment:
You don't have to change anything, but I think `check_func = x if cond else y` is more explicit :p